### ISSUE TITLE
Can't use metaprocessor commands in Jupyter

### ISSUE THREAD

**Issue by jarutis** [root-mirror/cling#11]
Hi,

Maybe I'm just doing something wrong... But if I try to use metaprocessor commands in Jupyter like 

```
.L libapophenia
```

I get an error 

```
input_line_22:2:2: error: expected expression
 .L libapophenia
```

All works well in cling launched from command line.

**Comment by punchEnbody** [comment]
have the same problem. Works in the regular CLI cling but in Jupyter none of the meta commands work. I've been trying to build it in various ways but no luck. Any suggestions?

**Comment by karies** [comment]
How does the current HEAD work for you (look for "Bind Jupyter to MetaProcessor instead of Interpreter."? I didn't actually try yet - busy with moving cling to the current llvm. But it compiles ;-)

**Comment by punchEnbody** [comment]
I see head is indeed updated. I'll try that. I have to admit I've been plodding through this. I had trouble with building originally (apparently regular make didn't do the Jupyter interpreter and the cmake directions were a bit off originally). Since then I've been successful with gcc 5 and clang 3.7. 

I was getting some confusion on python versions so I went to virtualenv. I hit a snag noted in a previous issue about shutil.which in python2.7 which I found today and I will try that fix.

However, no matter what I did, when I got cling to build the CLI did respond to metacommands the jupyter never does.

I'll let you know. Thanks for the help.

**Comment by punchEnbody** [comment]
First, thanks for your help. I'm working with python2.7 in a virtualenv. I had to make two changes to get it to run:
- make install crapped out on 
build/docs cmake_install.cmake with an error about missing file ocamldoc/html. I commented it out
- clingkernel.py, line 70, replaced the shutil.which with a try-except as suggested in issue #1 
- then did a pull and remade everything

 good news and bad news:

good news: Yes, metacommands now working. I can define a function for example

bad news: I get the following (in jupyter, not the kernel output on the CLI, which is different) on what looks like output

long l = 123;
std::cout << l << endl;
IncrementalExecutor::executeFunction: symbol 'cling_runtime_internal_throwIfInvalidPointer' unresolved while linking [cling interface function]!
